src/main/java/org/opensearch/neuralsearch/processor/AgenticQueryTranslatorProcessor.java

```diff
@@ -134,11 +162,21 @@ private void executeAgentAsync(
 
                 requestListener.onResponse(request);
             } catch (IOException e) {
-                log.error("Failed to parse agent response", e);
+                log.error(
+                    "Agentic search failed - Parse error - Agent ID: [{}], Query: [{}], Error: [{}]",
+                    agentId,
+                    agenticQuery.getQueryText(),
+                    e.getMessage()
+                );
                 requestListener.onFailure(e);
```

Comment:
QQ: if a listener is catching the exception then it will also post this a error.
Then why we need log.error seperately? Essentially, onFailure would throw this exception anyway